sty/statements/statements.dtx

```diff
@@ -1511,22 +1511,17 @@ sub defHelper{
   my $addr = LookupValue('defs');
   my $name = $whatsit->getArg(1);
   $name = $name->toString if $name;
-  if($defOption eq 'definiendum') {
-     $name = $whatsit->getArg(2) unless $name;
-     $whatsit->setProperty(name=>$name->toString);
-  } elsif($defOption eq 'defi') {
-     $name = $whatsit->getArg(2) unless $name;
-  } elsif($defOption eq 'adefi') {
-     $name = $whatsit->getArg(3) unless $name;
-  } elsif($defOption eq 'defii') {
-     $name = $whatsit->getArg(2)->toString.'-'.$whatsit->getArg(3)->toString unless $name;
-  } elsif($defOption eq 'adefii') {
-     $name = $whatsit->getArg(3)->toString.'-'.$whatsit->getArg(4)->toString unless $name;
-  } elsif($defOption eq 'defiii') {
-     $name = $whatsit->getArg(2)->toString.'-'.$whatsit->getArg(3)->toString.'-'.$whatsit->getArg(4)->toString unless $name;
-  } elsif($defOption eq 'adefiii') {
-     $name = $whatsit->getArg(3)->toString.'-'.$whatsit->getArg(4)->toString.'-'.$whatsit->getArg(5)->toString unless $name;
-   }
+  my %choose_Option = (
+                      definiendum => sub {$whatsit->setProperty(name=>$name->toString);
```

Comment:
The ```setProperty``` doesn't really belong there, you should move it out of the function and after ```$name``` is assigned to. It never hurts to have the name property set, so you can just add it for all cases.